network-mux/src/Network/Mux/Trace.hs

```diff
@@ -147,6 +152,11 @@ data MuxTrace =
     | MuxTraceStartedOnDemand MiniProtocolNum MiniProtocolDir
     | MuxTraceTerminating MiniProtocolNum MiniProtocolDir
     | MuxTraceShutdown
+#ifdef linux_HOST_OS
+    | MuxTraceTCPInfo StructTCPInfo Word16
+#else
+    | MuxTraceTCPInfo Word16
+#endif
```

Comment:
I haven't though about this, but with your original solution one was able to unify this constructor to
```
    | MuxTraceTCPInfo StructTCPInfo Word16
```

You can leave it as is, and I'll fix it next week (by providing two modules `Network.Mux.TCPInfo.Linux` and a public `Network.Mux.TCPInfo`).

Reply:
Sounds good.